components/brave_wallet/browser/json_rpc_service.cc

```diff
@@ -561,19 +551,24 @@ void JsonRpcService::RemoveChain(const std::string& chain_id,
 
 bool JsonRpcService::SetNetwork(const std::string& chain_id,
                                 mojom::CoinType coin,
+                                const absl::optional<url::Origin>& origin,
                                 bool silent) {
-  auto network_url = GetNetworkURL(prefs_, chain_id, coin);
-  if (!network_url.is_valid()) {
-    return false;
-  }
+  if (!origin) {
+    auto network_url = GetNetworkURL(prefs_, chain_id, coin);
+    if (!network_url.is_valid()) {
+      return false;
+    }
```

Comment:
Still need this?

Reply:
you meant `GetNetworkURL()` usage or writing into `kBraveWalletSelectedNetworks` as fallback?

Reply:
I see what you meant now, network_url is no longer needed and I will remove it.

Reply:
actually it is used as a way to validate if a chain_id is valid, I will use a new util function to do that.

Reply:
fixed in 29f3ed307238ab4611945e5ce51d42c44ff8d752